src/backend/backend.h

```diff
@@ -60,23 +60,55 @@ namespace snmalloc
 
     SNMALLOC_REQUIRE_CONSTINIT
     static inline FlatPagemap<MIN_CHUNK_BITS, PageMapEntry, PAL, fixed_range>
-      pagemap;
+      _pagemap;
+
+    struct Pagemap
+    {
+      /**
+       * Get the metadata associated with a chunk.
+       *
+       * Set template parameter to true if it not an error
+       * to access a location that is not backed by a chunk.
+       */
+      template<bool potentially_out_of_range = false>
+      SNMALLOC_FAST_PATH static const MetaEntry& get_meta_data(address_t p)
+      {
+        return _pagemap.template get<potentially_out_of_range>(p);
+      }
+
+      /**
+       * Set the metadata associated with a chunk.
+       */
+      SNMALLOC_FAST_PATH
+      static void set_meta_data(address_t p, size_t size, MetaEntry t)
```

Comment:
These things should probably take a `LocalState*` as an argument, if the caller has a valid `LocalState`.

Reply:
Sorry, can you expand on the rationale?

Reply:
In configurations where all of the state is in LocalState and the lazy init path is not used, we have no way of getting at the pagemap.  Or, for example, if accessing the pagemap is expensive then I might add a local cache of some pagemap values in LocalState.  This is the general pattern for everything in `Backend`: it either takes a `LocalState&` if the callers are all guaranteed to have a `LocalState` or a `LocalState*` if some of them might not.  Unless you can provide a rationale that no implementation of a back end will ever want a LocalState in this interface then we should optionally provide one.

Reply:
Ah, right; thank you.  I've taken a stab at that as the last commit in the series (as of this writing, of course).